wms-maps-backgrounds/js/wms.js

```diff
@@ -1,22 +1,33 @@
 (function() {
-    const pluginSettings = window.dkuAppConfig.loadedPlugins.find(d => d.id === 'wms-maps-backgrounds').params;
+    const pluginDesc = window.dkuAppConfig.loadedPlugins.find(d => d.id === 'wms-maps-backgrounds');
+    
+    DataikuAPI.plugins.get("wms-maps-backgrounds").success(function(data) {
+        if (!data || !data.settings) {
+            console.info("WMS Plugin: no settings, bailing out");
+            return;
+        }
+        var wmsList = [];
+        if (data.settings.presets && data.settings.presets.length > 0) {
+            // new style: with parameter sets
+            console.info("Registering WMS map backgrounds from presets");
+            wmsList = data.settings.presets.map(p => [p.config.url, p.config.layerId, p.name]);
+        } else if (data.settings.config && data.settings.config.wmsResources && data.settings.config.wmsResources.length > 0) {
+            // old style, everything in a textarea
```

Comment:
Why do we need this if plugin params are removed in this version?

Reply:
without this, updating breaks the existing plugin installs because the place settings are stored has changed

Reply:
Does that mean that upon migration existing settings will be copied from the old place to the new place?